tombstones.go

```diff
@@ -0,0 +1,90 @@
+package zoekt
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+)
+
+// TombstoneEnabled returns true if a file "RIP" is present in dir.
+func TombstonesEnabled(dir string) bool {
+	_, err := os.Stat(filepath.Join(dir, "RIP"))
+	return err == nil
+}
+
+var mockRepos []*Repository
+
+// SetTombstone idempotently sets a tombstone for repoName in .meta.
+func SetTombstone(shardPath string, repoName string) error {
+	var repos []*Repository
+	var err error
+
+	if mockRepos != nil {
+		repos = mockRepos
+	} else {
+		repos, _, err = ReadMetadataPath(shardPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, repo := range repos {
+		if repo.Name == repoName {
+			repo.Tombstone = true
+		}
+	}
+
+	dest := shardPath + ".meta"
+	fn, err := jsonMarshalTmpFile(repos, dest)
+	if err != nil {
+		return err
+	}
+
+	err = os.Rename(fn, dest)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// jsonMarshalFileTmp marshals v to the temporary file p + ".*.tmp" and
+// returns the file name.
+//
+// Note: .tmp is the same suffix used by Builder. indexserver knows to clean
+// them up.
+func jsonMarshalTmpFile(v interface{}, p string) (_ string, err error) {
```

Comment:
might as well include the rename part here? then you can also remove the tmp file in case of rename failure.

Reply:
done